SignalUtilitiesKit/Utilities/UIColor+Extensions.swift

```diff
@@ -0,0 +1,43 @@
+// Copyright © 2022 Rangeproof Pty Ltd. All rights reserved.
+
+import UIKit.UIColor
+
+public extension UIColor {
+    struct HSBA {
+        public var hue: CGFloat = 0
+        public var saturation: CGFloat = 0
+        public var brightness: CGFloat = 0
+        public var alpha: CGFloat = 0
+
+        public init?(color: UIColor) {
+            // Note: Looks like as of iOS 10 devices use the kCGColorSpaceExtendedGray color
+            // space for grayscale colors which seems to be compatible with the RGB color space
+            // meaning we don'e need to check 'getWhite:alpha:' if the below method fails, for
+            // more info see: https://developer.apple.com/documentation/uikit/uicolor#overview
+            guard color.getHue(&hue, saturation: &saturation, brightness: &brightness, alpha: &alpha) else {
+                return nil
+            }
+        }
+    }
+
+    var hsba: HSBA? { return HSBA(color: self) }
+
+    // MARK: - Functions
+
+    func toImage() -> UIImage {
+        let bounds: CGRect = CGRect(x: 0, y: 0, width: 1, height: 1)
+        let renderer: UIGraphicsImageRenderer = UIGraphicsImageRenderer(bounds: bounds)
+
+        return renderer.image { rendererContext in
+            rendererContext.cgContext.setFillColor(self.cgColor)
+            rendererContext.cgContext.fill(bounds)
+        }
+    }
+
+    func darken(by percentage: CGFloat) -> UIColor {
+        guard percentage != 0 else { return self }
+        guard let hsba: HSBA = self.hsba else { return self }
+
+        return UIColor(hue: hsba.hue, saturation: hsba.saturation, brightness: (hsba.brightness - percentage), alpha: hsba.alpha)
+    }
+}
```

Comment:
Can we use RGB or existed colors in SessionUIKit? Just want to keep the codebase simple.

Reply:
So taking a look at the `Colors` in SessionUIKit there isn't a white which doesn't change for light/dark mode which is needed for this button (There is the `CirclePlay` button which is used for video attachments but I've used the separate `Play` and `Pause` buttons with a generated background here to try and keep the logic consistent with the old code which was using [audio_play_black_large](https://github.com/oxen-io/session-ios/blob/16c2639bf22c4f61266144580f3d059dd17e6bb3/Session/Meta/Images.xcassets/audio_play_black_large.imageset/audio_play_black_large@3x.png) & [audio_pause_black_large](https://github.com/oxen-io/session-ios/blob/16c2639bf22c4f61266144580f3d059dd17e6bb3/Session/Meta/Images.xcassets/audio_pause_black_large.imageset/audio_pause_black_large@3x.png) before they were removed in https://github.com/oxen-io/session-ios/commit/e6b91fdbeea41ce90eae2ec6e299f676cde3362a resulting in the crash)

<img width="200" alt="Screen Shot 2022-01-14 at 1 34 56 pm" src="https://user-images.githubusercontent.com/15862619/149441867-80c06670-782d-41b7-9212-5dd670fa8912.png">

Happy to change it to not have the 'pause' state to be consistent with video attachments if you'd prefer but I think having the pause button here as well is a better experience